debian/consul.upstart

```diff
@@ -8,13 +8,15 @@ respawn
 script
   # Make sure to use all our CPUs, because Consul can block a scheduler thread
   export GOMAXPROCS=`nproc`
+  USER="consul"
+  OPTS="agent -config-dir="/etc/consul.d ${CONSUL_FLAGS}"
 
   # Allow overriding env vars in /etc/default/consul
   if [ -f "/etc/default/consul" ]; then
     . /etc/default/consul
   fi
 
-  exec /usr/bin/consul agent \
```

Comment:
Hi, thanks for the review! Indeed, that make sense, I'll modify this PR in this direction.